### PR TITLE
add test for malformed websockets

### PR DESCRIPTION
**Pull request by cjrkoa**
Related #784

### CODE REVIEW

**Comment by dyc3** [comment]
The key words for this issue are in the logs in #784 :
```
Invalid WebSocket frame: RSV2 and RSV3 must be clear
```
RSV2 and RSV3 are bits that can be set in the websocket protocol. 

You should take a look at the code in `tools/ws-fuzz`, that was the code I wrote when trying to fix this issue originally.

File: crates/harness/src/test_runner.rs [hunk 1]
```diff
@@ -43,15 +43,20 @@ impl TestRunner {
     }
 
     /// Kill the balancer and start a new one with the same configuration.
-    pub async fn restart_balancer(&mut self) {
+    pub async fn restart_balancer(&mut self) -> bool {
```
Comment (dyc3) [inline crates/harness/src/test_runner.rs:46]: This just makes this function more complex. Is there any reason why you can't do this in a new function?

File: crates/harness-tests/src/routing.rs [hunk 2]
```diff
@@ -328,28 +329,18 @@ async fn should_prioritize_same_region_ws(ctx: &mut TestRunner) {
...
 async fn test_malformed_header_rsv2_rsv3(ctx: &mut TestRunner) {
-    // What I'm thinking I need to do for this at the most basic level:
-    // 1. Create a new client
-    // 2. Create a WebSocket with a malformed header
-    // 3. Link it to the client
-    // 4. Try to connect to OTT through the balancer
-    // 5. Test for no crash?
-    
-    let mut c = Client::new(ctx).unwrap();
+    let mut client = ClientBuilder::new("ws://localhost:8081/api/room/test").connect_insecure()?; //no method named `connect_insecure` found for enum `Result` in the current scope method not found in `Result<ClientBuilder<'_>, ParseError>`
```
Comment (dyc3) [inline crates/harness-tests/src/routing.rs:332]: This url uses the wrong port, you should be using `ctx.url()` to create this url.

Are these comments here for me? These are just compiler error messages.

File: crates/harness-tests/src/routing.rs [hunk 3]
```diff
@@ -328,28 +329,18 @@ async fn should_prioritize_same_region_ws(ctx: &mut TestRunner) {
...
+    let dataframe = DataFrame {
         finished: true,
         reserved: [false, true, true],
         opcode: websocket::dataframe::Opcode::Text,
         data: "{\"action\":\"auth\", \"token\":\"foo\"}"
             .to_string()
             .into_bytes(),
     };
+    
+    client.send_dataframe(&dataframe).await?; // the `?` operator can only be used in an async function that returns `Result` or `Option` (or another type that implements `FromResidual`) the trait `FromResidual<_>` is not implemented for `()`
```
Comment (dyc3) [inline crates/harness-tests/src/routing.rs:343]: same for this comment

Comment (cjrkoa) [reply crates/harness-tests/src/routing.rs:343]: These were just the errors I was getting when I left off

**Comment by dyc3** [comment]
You should rebase/merge main, I made a couple changes in #1270 to make this easier for you.

**Pull request closed without merging**